src/python/pants/backend/experimental/jvm/resolve/coursier_fetch_integration_test.py

```diff
@@ -0,0 +1,306 @@
+    # in nearly a decade, it's still brittle in a unit test to rely on an externally fetched file staying stable.
+
+    classpath_entry = rule_runner.request(
+        ResolvedClasspathEntry,
+        [
+            CoursierLockfileEntry(
+                coord=MavenCoord(coord="junit:junit:4.13.2"),
+                file_name="junit-4.13.2.jar",
+                direct_dependencies=MavenCoordinates(
+                    [MavenCoord(coord="org.hamcrest:hamcrest-core:1.3")]
+                ),
+                dependencies=MavenCoordinates([MavenCoord(coord="org.hamcrest:hamcrest-core:1.3")]),
+                file_digest=FileDigest(
+                    fingerprint="8e495b634469d64fb8acfa3495a065cbacc8a0fff55ce1e31007be4c16dc57d3",
+                    serialized_bytes_length=384581,
+                ),
+            )
+        ],
+    )
+    assert classpath_entry.coord == MavenCoord(coord="junit:junit:4.13.2")
+    assert classpath_entry.file_name == "junit-4.13.2.jar"
+    file_digest = rule_runner.request(
+        FileDigest, [ExtractFileDigest(classpath_entry.digest, "junit-4.13.2.jar")]
+    )
+    assert file_digest == FileDigest(
+        fingerprint="8e495b634469d64fb8acfa3495a065cbacc8a0fff55ce1e31007be4c16dc57d3",
+        serialized_bytes_length=384581,
+    )
+
+
+def test_fetch_one_coord_with_bad_fingerprint(rule_runner: RuleRunner) -> None:
+    # TODO: Use a mocked Process to test these kinds of details.  While this version hasn't changed
+    # in nearly a decade, it's still brittle in a unit test to rely on an externally fetched file staying stable.
+
+    expected_exception_msg = (
+        r".*?CoursierError:.*?Coursier fetch for .*?hamcrest.*? succeeded.*?"
+        r"66fdef91e9739348df7a096aa384a5685f4e875584cce89386a7a47251c4d8e9.*?"
+        r"did not match.*?ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff"
+    )
+    with pytest.raises(ExecutionError, match=expected_exception_msg):
+        rule_runner.request(
+            ResolvedClasspathEntry,
+            [
+                CoursierLockfileEntry(
+                    coord=MavenCoord(coord="org.hamcrest:hamcrest-core:1.3"),
+                    file_name="hamcrest-core-1.3.jar",
+                    direct_dependencies=MavenCoordinates([]),
+                    dependencies=MavenCoordinates([]),
+                    file_digest=FileDigest(
+                        fingerprint="ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff",
+                        serialized_bytes_length=45024,
+                    ),
+                )
+            ],
+        )
+
+
+def test_fetch_one_coord_with_bad_length(rule_runner: RuleRunner) -> None:
+    # TODO: Use a mocked Process to test these kinds of details.  While this version hasn't changed
+    # in nearly a decade, it's still brittle in a unit test to rely on an externally fetched file staying stable.
```

Comment:
Maybe make a single TODO at the top of the file. It'd be great to have an issue tracking it too, so that you can do `# TODO(#xxx)`.